src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -460,10 +478,13 @@ private ReadEndsForMarkDuplicates buildReadEnds(final SAMFileHeader header, fina
      *
      * @return an array with an ordered list of indexes into the source file
      */
-    private void generateDuplicateIndexes(final boolean useBarcodes) {
+    private void generateDuplicateIndexes(final boolean useBarcodes, final boolean indexOpticalDuplicates) {
         // Keep this number from getting too large even if there is a huge heap.
-        final int maxInMemory = (int) Math.min((Runtime.getRuntime().maxMemory() * 0.25) / SortingLongCollection.SIZEOF,
-                (double) (Integer.MAX_VALUE - 5));
+        int maxInMemory = (int) Math.min((Runtime.getRuntime().maxMemory() * 0.25) / SortingLongCollection.SIZEOF, (double) (Integer.MAX_VALUE - 5));
```

Comment:
I think it will be clearer if maxInmemory remains final and you use maxinMemory/2 as the input for the opticalDupIndex. BTW, can you comment on why you need half the amount?